plugin-overlay/src/main/java/com/mapbox/maps/plugin/overlay/MapOverlayPluginImpl.kt

```diff
@@ -169,35 +170,37 @@ class MapOverlayPluginImpl : MapOverlayPlugin {
       var size = queue.size
       while (size > 0) {
         val subMapRect = queue.pollFirst()
-        if (it.isOverLap(subMapRect)) {
-          // insert the new subMapRect
-          if (it.top - subMapRect.top < subMapRect.bottom - it.bottom) {
-            // overlay rect is on the top part of MapRect, create a new MapRect with the bottom part
-            queue.offerLast(subMapRect.updateTop(it.bottom))
-          } else if (it.top - subMapRect.top > subMapRect.bottom - it.bottom) {
-            // overlay rect is on the bottom part of MapRect, create a new MapRect with the top part
-            queue.offerLast(subMapRect.updateBottom(it.top))
-          } else {
-            // overlay rect is on the center of MapRect, create a new MapRect with both the top part and bottom part
-            queue.offerLast(subMapRect.updateTop(it.bottom))
-            queue.offerLast(subMapRect.updateBottom(it.top))
-          }
+        subMapRect?.let {
+          if (it.isOverLap(subMapRect)) {
```

Comment:
`it` is confusing here, can you change to 
`queue.pollFirst()?.let { subMapRect->`